ts/Core/Renderer/SVG/SVGElement.ts

```diff
@@ -474,13 +479,21 @@ class SVGElement implements SVGElementLike {
             vAlignFactor = 2;
         }
         if (vAlignFactor) {
-            y += ((box as any).height - ((alignOptions as any).height || 0)) /
+            y += ((box as any).height - (alignOptions.height || 0)) /
                 vAlignFactor;
         }
         attribs[alignByTranslate ? 'translateY' : 'y'] = Math.round(y);
 
         // Animate only if already placed
         if (redraw) {
+
+            // Right- and center-aligned labels should not animate when we're
+            // only changing the text (#20965)
+            if (!animateX) {
+                this.attr({
+                    [alignByTranslate ? 'translateX' : 'x']: x
```

Comment:
Can we use computed property names?

Reply:
Yes, apparently [supported since 2016](https://caniuse.com/?search=computed%20property%20names) in the major browsers.